webui/react/src/hooks/useSettings.ts

```diff
@@ -44,7 +44,7 @@ const settingsToQuery = <T>(config: SettingsConfig<T>, settings: Settings) => {
 
   if (qParams.toString().length !== 0) {
     for (const param of qParams.keys()) {
-      if (settings[param] === undefined) {
+      if (!(param in settings)) {
```

Comment:
I believe this change is the necessary for the fix. This section of the code is to add non-setting params back to the url, so suppose we want to reset a setting `foo`, with current query `foo=bar`. When we pass `{foo: undefined}` to reset, this section of the code would add `foo=bar` back to the query without the change.

Reply:
good point, but given that we don't otherwise check for overlap and just call `retVal.set` on values in the settings object, i think it'd overwrite anyways. at any rate, i think the overall effect is the same.